collectors/python.d.plugin/nvidia_smi/nvidia_smi.chart.py

```diff
@@ -325,14 +350,24 @@ def get_data(self):
             return None
 
         last_data = self.poller.data()
+        if not last_data:
+            return None
 
         parsed = self.parse_xml(last_data)
         if parsed is None:
             return None
 
         data = dict()
         for idx, root in enumerate(parsed.findall('gpu')):
-            data.update(GPU(idx, root).data())
+            gpu = GPU(idx, root)
+            data.update(gpu.data())
+            ps = gpu.processes() or []
+
+            chart = self.charts['gpu{0}_{1}'.format(gpu.num, PROCESSES_MEM)]
+            for p in ps:
+                dim_name = 'gpu{0}_process_mem_{1}'.format(gpu.num, p['pid'])
+                if dim_name not in chart:
+                    chart.add_dimension([dim_name, '{0} {1}'.format(p['pid'], p['process_name'])])
```

Comment:
Do we need this on every call?

Reply:
I think so. Because processes can be created at any time. Every `get_data` may contains newly created processes

Reply:
Hm, then we need to delete stale processes from the chart. We can add it later tho. Do you know how many processes can be created during lifetime? I am trying to understand is cleanup is critical or not.

Reply:
Sorry for the delay. I'm using this to monitor GPUs running machine learning training processes. Typically, each GPU will run 1-4 processes, and each process will run for several hours. I'm going to add this function tomorrow.

Reply:
use https://github.com/netdata/netdata/pull/5795